src/main/java/org/hmhb/organization/DefaultOrganizationService.java

```diff
@@ -0,0 +1,118 @@
+package org.hmhb.organization;
+
+import javax.annotation.Nonnull;
+import javax.transaction.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.codahale.metrics.annotation.Timed;
+import org.apache.commons.lang3.StringUtils;
+import org.hmhb.audit.AuditHelper;
+import org.hmhb.exception.organization.CannotDeleteOrganizationWithProgramsException;
+import org.hmhb.exception.organization.OrganizationNameRequiredException;
+import org.hmhb.exception.organization.OrganizationNotFoundException;
+import org.hmhb.program.ProgramDao;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class DefaultOrganizationService implements OrganizationService {
```

Comment:
This code is mostly identical to the code it replaces.  Just a different domain object name and validation logic.